### ISSUE TITLE
Add Entry in Autofill

### ISSUE THREAD

**Issue by yacob841** [keepassium/KeePassium#87]
I mentioned this in reddit awhile ago but figured I would post it in here.

If we are on a website that we do not have an account for, it would be nice if there was a + button in the autofill popup that would either allow us to add an entry if autofill extension will allow it or it would open up KeePassium to the add entry page allowing us to add an entry. It gets tedious having to close safari, open KeePassium, add entry, close KeePassium, reopen safari, and then use autofill to fill the new password.

Or better yet, if you had a “default” username the + could auto create an entry with the current url, best guess name, default username, and give a prompt for password options and then when selected it will autofill on the page.

**Comment by samsam-rolon** [comment]
Very Intrested in !

**Comment by loeffelpan** [comment]
Just tested out KeePassium and missed exactly this feature. Any updates on this.

I‘m very interested, too.
It‘s the ### only feature I‘m missing.

**Comment by keepassium** [comment]
Yep, it is in the list. There are a few higher-priority tasks in the queue, but eventually I'll get to this one, too.

**Comment by ospring** [comment]
I am very interested too !

**Comment by waynezhang** [comment]
This would be the last missing feature for KeePassium! :laughing

**Comment by DevertNet** [comment]
+1 this would be so helpful!

**Comment by eingutertag** [comment]
+1 this would be so helpful for me!

**Comment by loeffelpan** [comment]
This is now on the list for a long time.
Any news?
It will increase the my wifes acceptance of KeePass a lot. She saya it is complicated to switch apps while registering to online-shops.
Would love to see this feature in KeePassium soon.

**Comment by keepassium** [comment]
No news yet, I'm afraid… 

At a first glance, this feature looks straightforward: add a screen with prefilled entry data, and a few of buttons to save/edit/cancel. The challenge, however, is that AutoFill works in very [tight memory bounds](https://keepassium.com/articles/autofill-memory-limits/) which are barely enough to _load_ the database. Encrypting the DB would double or triple the memory consumption, so the system is very likely to terminate the AutoFill before it finishes saving the file.

As a workaround, AutoFill could temporarily save the new entry somewhere else and then launch the main app to encrypt and save the main database. But this would be so clumsy and ugly that I just self-censor such a solution. And for the lack of other ideas, a bit stuck here...

**Comment by loeffelpan** [comment]
Sound not solveable 😔 So sad. I would have loved this feature.

**Comment by keepassium** [comment]
Oh, it is solvable, it's just hard to make it reliable. Of course, there is always the option of showing "Your database is too large, so it will be read-only", but I'd like to avoid this if possible.

**Comment by tberta** [comment]
Here is a suggestion to ease the entry creation process but I don’t know if it is possible:
when autofill has no entry for a given url, display a button that open the full KeePassium app. Right after unlocking, display a create entry dialog with some fields already pre-filled (url, title)
Is it possible?

**Comment by keepassium** [comment]
@tberta, this is a good point, thank you!

**Comment by B1os1618** [comment]
A lack of mandatory functionality

**Comment by uberfunk** [comment]
Even with all the explanations of limited memory, I still don't understand how my old 1Password app (7.10.2) is able to do this but Keepassium can't.
![IMG_1595](https://github.com/keepassium/KeePassium/assets/45521405/18c33a85-f32b-4aef-b178-9b4871c481e5)

**Comment by keepassium** [comment]
>  I still don't understand how my old 1Password app (7.10.2) is able to do this but Keepassium can't.

Just like an old iPhone 4 can fit in a pocket but iPhone 15 Pro Max can't… A suitcase would pass through most doors; a shipping container  — not so much.

It all depends on the size. If your database is reasonably small (couple of MBs), it will work just fine. If it is a couple _dozens_ of MB, it shall not pass™ 

We don't know how 1Password works behind the scenes, but as far as I know, their only user-facing encryption parameter is the master password. Oh, and I really hope 1Password data format is not a compressed XML file and they don't have to load the whole thing in the memory.

**Comment by Doc-Steve** [comment]
Would the process Strongbox uses be an option for Keepassium? How it works there: when you create a new entry in the autofill menu the new entry is cached and not written to the database. There is warning that you need to open the app later to include the cached entry in your database.

**Comment by keepassium** [comment]
@Doc-Steve , yep, this is the current approach. 

I avoided it for years, because this is a kludge that breaks immediate file sync. (KeePassium AutoFill does fetch remote DBs, Strongbox AutoFill seems to load cached files instead.) So the user may end up with several new entries that _appear_ to be in database, but really exist only in phone's keychain until the main app opens.

But this seems to be the only reliable way to modify DBs from within AutoFill. And that fact that it's much easier to implement than proper encryption in a memory-restricted process also helps :)

**Comment by MahirAbbas** [comment]
Yes. A much easier option is what @St0rmEye-GH mentioned. Even just a manual intervention, where you cache the entry name, and the caller ID, and then upon next load, the user gets presented with them, and can manually merge/copy in the URL. possibly 2 fields (one for Apps, the other for website URLs).

**Comment by keepassium** [comment]
Released in v 2.4 🥳
